ArduCopter/radio.cpp

```diff
@@ -180,7 +180,7 @@ void Copter::set_throttle_zero_flag(int16_t throttle_control)
     // and we are flying. Immediately set as non-zero
     if ((!ap.using_interlock && (throttle_control > 0) && !SRV_Channels::get_emergency_stop()) ||
         (ap.using_interlock && motors->get_interlock()) ||
-        ap.armed_with_switch) {
+        ap.armed_with_switch || ap.air_mode == AirModeEnabled) {
```

Comment:
```suggestion
        ap.armed_with_switch || (rc().find_channel_for_option(RC_Channel::AUX_FUNC::AIRMODE) && rc().find_channel_for_option(RC_Channel::AUX_FUNC::AIRMODE)->get_channel_pos() == aux_switch_pos_t::HIGH)) {
```

This looks a bit clunky; we could probably have a convenience method on `rc()` to do the same thing - `bool rc().aux_switch_pos_is(aux_switch_pos_t value)` which returns false if the function isn't assigned.

Note that that `ap.using_interlock` is actually what I'm proposing here, just going via some global state.

Reply:
find_channel_for_option() scans through all RC channels every time and this is called all the time. Doesn't seem a great thing to do at this point in the code.

Reply:
I'm going to disagree with the suggestion :)